packages/dev/docs/src/DocSearch.js

```diff
@@ -0,0 +1,349 @@
+  const [loadingState, setLoadingState] = useState(null);
+  const [predictions, setPredictions] = useState(null);
+  const [suggestions, setSuggestions] = useState([]);
+
+  const searchAutocompleteRef = useRef();
+
+  useEffect(() => {
+    // Focus search input when user presses Ctrl/Cmd + K
+    let handleKeyDown = (e) => {
+      if ((e.ctrlKey || e.metaKey) && e.key === 'k') {
+        e.preventDefault();
+        searchAutocompleteRef.current.focus();
+      }
+    };
+
+    document.addEventListener('keydown', handleKeyDown);
+
+    return () => {
+      document.removeEventListener('keydown', handleKeyDown);
+    };
+  }, []);
+
+  let updatePredictions = ({hits}) => {
+    setPredictions(hits);
+
+    const groupedBySection = groupBy(hits, (hit) => sectionTitlePredicate(hit));
+    let sections = [];
+    for (const [title, hits] of Object.entries(groupedBySection)) {
+      const items = Object.values(
+        groupBy(hits, (hit) => hit.hierarchy.lvl1)
+      )
+      .reverse()
+      .map(
+        groupedHits =>
+        groupedHits.map((hit) => {
+          const hierarchy = hit.hierarchy;
+          const objectID = hit.objectID;
+          const docsearchParent = hit.type !== 'lvl1' &&
+            groupedHits.find(
+              (siblingItem) =>
+                siblingItem.type === 'lvl1' &&
+                siblingItem.hierarchy.lvl1 ===
+                  hit.hierarchy.lvl1
+            );
+
+          return {
+            key: objectID,
+            hit,
+            hierarchy,
+            docsearchParent,
+            textValue: hit.type === 'content' ? hit[hit.type] : hierarchy[hit.type]
+          };
+        }
+      ));
+
+      sections.push({title, items: items.map((item) => item[0])});
+    }
+    let titles = Object.values(sectionTitles);
+    sections = sections.sort((a, b) => titles.indexOf(a.title) < titles.indexOf(b.title) ? -1 : 1);
+    let newSuggestions = sections.map((section, index) => ({key: `${index}-${section.title}`, title: section.title, children: section.items}));
+    newSuggestions.push({
+      key: 'algolia-footer',
+      title: 'Search by',
+      children: [
+        {
+          key: 'algolia-footer-logo',
+          textValue: 'Algolia'
+        }
+      ]
+    });
+    setSuggestions(newSuggestions);
+    setLoadingState(null);
+  };
+
+  let onInputChange = (query) => {
+    setSearchValue(query);
+    if (!query && predictions) {
+      setPredictions(null);
+      setSuggestions([]);
+      setLoadingState(null);
+      return;
+    }
+    setLoadingState('filtering');
+    searchIndex
+      .search(
+        query,
+        searchOptions
+      )
+      .then(updatePredictions);
+  };
+
+  let onSubmit = (value, key) => {
+    if (key === 'algolia-footer-logo') {
+      window.open('https://www.algolia.com/ref/docsearch/?utm_source=react-spectrum.adobe.com&amp;utm_medium=referral&amp;utm_content=powered_by&amp;utm_campaign=docsearch', '_blank');
```

Comment:
mega nit: mind reseting the input value here so "algolia" doesn't persist as the input text?

```suggestion
    if (key === 'algolia-footer-logo') {
      window.open('https://www.algolia.com/ref/docsearch/?utm_source=react-spectrum.adobe.com&amp;utm_medium=referral&amp;utm_content=powered_by&amp;utm_campaign=docsearch', '_blank');
      setSearchValue('');
```

Reply:
I think onSubmit fires before the input is filled so I'll have to use a different method.

Reply:
It looks like the above only works when hitting enter to select Algolia but not when clicking on the option, strange

Reply:
Weird that the field clears on clicking it, even without the code you suggested above. Definitely seems like a SearchAutocomplete bug. 

The only solution I can think of now is to refresh the page with `location.reload()`, but that seems like overkill.

Reply:
Yeah, I've been doing some digging but haven't come up with a good alternative. Kinda feels like there is something broken in useComboBoxState with resetInputValue conflicting with the user's call of setSearchValue, but also I think SearchAutocomplete should perhaps do away with the concept of tracking the selectedKey since selecting an option would perform some kind of operation (maybe it should control the selectedKey to `null` instead of setting it to undef). For now the current experience is fine

Reply:
Actually, this feels a little better as a hack:
`searchAutocompleteRef.current.UNSAFE_getDOMNode().querySelector('[role="button"]').click();`

Reply:
pretty gross haha but I'm fine with it for now.